apis/ml-projects/ml-projectsapi.yaml

```diff
@@ -0,0 +1,1863 @@
+    - The ML-Project Service Apis perform operations related to create and manage user project, project template and reports. 
+    
+    - The URL for ML-Project Service API(s) is `{context}/mlprojects/v1`
+
+    - <b>Note:</b> These resources can be used in other services 
+  contact: {}
+  version: '1.0'
+servers:
+- url: https://staging.open-sunbird.org/api
+  variables: {}
+paths: 
+  /userProjects/mlprojects/v1/details/{projectId}:
+    get:
+      tags:
+      - userProjects APIs
+      summary: Get Project Details.
+      description: |-
+        This API is used to get the Project Details. This api will return the project details based on the role and location Id.
+          - The endpoint for **UserProjects details** is `/userProjects/mlprojects/v1/details`
+          - It is mandatory to provide values for parameters marked with *. 
+          - Mandatory fields cannot be null or empty.
+
+      operationId: details
+      parameters:
+        - name: X-Authenticated-User-Token
+          in: header
+          description: Represents the authorized User ID to execute the API.
+          required: true
+          schema:
+            type: string
+        - name: Authorization
+          in: header
+          description: >-
+            To make use of the API, you require authorization. Raise a request
+            to the administrator for the use of the API. You will receive the
+            authorization key. Specify the key received, here.
+          required: true
+          schema:
+            type: string
+        - name: projectId
+          in: path
+          description: Please append a valid project Id To the Request URL
+          required: true
+          schema:
+            type: string
+          example: 5f731631e8d7cd3b88ac0659
+      responses:
+        200:
+          description: "Project Details Successful Response"
+          content:
+            application/json:
+              schema:
+                  $ref: '#/components/schemas/userprojectsapiDetailsResponse'
+        400:
+          description: 'BAD REQUEST. The operation failed ! The possible reason for failure is that you may have missed providing input for a mandatory parameter.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/400ErrorProjectIdMissing'
+        500:
+          description: "'INTERNAL SERVER ERROR. We track these errors automatically and try to set it right at the earliest. Try refreshing the page. If the problem persists contact us at info@sunbird.org.'"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/500Error'
+      deprecated: false
+      security:
+        - bearer: []
+  /userProjects/mlprojects/v1/add:
+    post:
+      tags:
+      - userProjects APIs
+      summary: Add project.
+      description: |-
+        This API is associated with creating project on the Sunbird Platform.
```

Comment:
@priyanka-TL  is this the right definition of this API "This API creates projects on top user object"?